src/ruby/travelling-salesman.rb

```diff
@@ -12,11 +12,10 @@ def initialize(nodes, startMatrix)
     ##
     # shortest path function
     # need to fix matrix override
-    public
-    def shortestPath 
-        for node in @nodes
-            addOption(0, node)
-            addNextNode(0, node)
+    def shortestPath
+        @nodes.each_with_index do |_, nodeIndex|
```

Comment:
Having a variable in a block named `_` is a convention which means you don't really give a shit about the value and you won't be using it. You could easily have just used the `times` method and used the variable from that but I figured this method read a little better.